hw/ip/adc_ctrl/doc/_index.md

```diff
@@ -1,34 +1,215 @@
 ---
 
 # Overview
 
-
+This document specifies the ADC controller IP functionality.
+This IP block implements control and filter logic for an analog block that implements a dual ADC.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for integration overview within the broader top level system
 
 ## Features
 
+The IP block implements the following features:
 
+- Register interface to dual ADC analog block
+- Support for 2 ADC channels
+- Support for 8 filters on the values from the channels
+- Support ADCs with 10-bit output (two reserved bits in CSR)
+- Support for debounce timers on the filter output
+- Uses slow clock to enable use while the part is sleeping
+- Low power periodic scan mode for monitoring ADC channels
 
 ## Description
 
+The ADC controller is a simple front-end to an analog block that allows filtering and debouncing of the analog signals
+
+## Compatibility
 
+The ADC controller programming interface is not based on any existing interface
 
 # Theory of Operation
 
+The block diagram shows a conceptual view of the ADC controller state machine and filters
+
 ## Block Diagram
 
+![ADC_CTRL Block Diagram](adc_overview.svg "image_tooltip")
 
 ## Hardware Interface
 
 {{< incGenFromIpDesc "../data/adc_ctrl.hjson" "hwcfg" >}}
 
 ## Design Details
 
+## Sampling state machine
 
-# Programmers Guide
+The state machine that takes ADC samples follows a very simple pattern:
+
+1. *Power up ADC*: The controller issues the power up command to the ADC.
+
+2. *Wait for ADC turn on*: The controller waits for the number of clock cycles programmed in {{< regref "adc_pd_ctl.pwrup_time" >}} which should be set to match the ADC power up delay.
+
+3. *Take sample Channel 0*: The ADC is requested to sample channel 0. When the ADC signals complete the value is stored in {{< regref "adc_chn_val[0].adc_chn_value" >}}. Note that the time taken in this step depends on the properties of the ADC.
+
+4. *Take sample Channel 1*: The ADC is requested to sample channel 1. When the ADC signals complete the value is stored in {{< regref "adc_chn_val[1].adc_chn_value" >}}. Note that the time taken in this step depends on the properties of the ADC.
+
+5. *Evaluate Filters*: The filters are evaluated and debounce logic applied (see [next section](#filters-and-debounce).
+
+6. *Scan type check*: At this point if the {{< regref "adc_pd_ctl.lp_mode" >}} bit is clear scanning continues at step (3).
+   If the bit is set the next step depends on how many samples have hit the filters.
+   If more than {{< regref "adc_lp_sample_ctl.lp_sample_cnt" >}} samples have hit then continuous scanning continues at step (3) else periodic scanning will continue at the next step (7).
+
+7. *Power off ADC*: The controller issues the power down command to the ADC.
+
+8. *Wait sleep time*: the controller will wait for the next sample timer to timeout before restarting at step (1).
```

Comment:
Nit: Capitalize *T*he for consistency with other bullet points

Reply:
Sure